linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_faillock_deny/oval/shared.xml

```diff
@@ -1,222 +1,290 @@
+    <ind:filepath operation="pattern match">^/etc/pam.d/system-auth$</ind:filepath>
+    <ind:pattern operation="pattern match"
+                 var_ref="var_accounts_passwords_pam_faillock_deny_pam_faillock_auth_regex"/>
     <ind:instance datatype="int" operation="equals">1</ind:instance>
   </ind:textfilecontent54_object>
 
-  <!-- step 5 -->
-  <local_variable id="var_accounts_passwords_pam_faillock_preauth_default_lines_regex_system-auth" datatype="string" version="1" comment="Regex containing skipped lines">
-    <concat>
-      <literal_component datatype="string">pam_unix(?:.*[\n](?:.*[\n]){</literal_component>
-      <object_component item_field="subexpression" object_ref="object_accounts_passwords_pam_faillock_lines_value_system-auth" />
-      <literal_component datatype="string">})(?:.*[\n])*auth.*pam_faillock.so[\s]+[^\n]*deny=([0-9]+)</literal_component>
-    </concat>
-  </local_variable>
-
-  <!-- end of system-auth handling, and beginning of password-auth handling -->
-  <!-- step 1 and 3 test -->
-  <ind:textfilecontent54_test id="test_accounts_passwords_pam_faillock_numeric_default_check_password-auth"
-  check="all" check_existence="all_exist"
-  comment="Checks if pam_faillock authfail is hit even if pam_unix skips lines by defaulting, and also authfail deny value" version="2">
-    <ind:object object_ref="object_accounts_passwords_pam_faillock_when_lines_skipped_password-auth" />
-    <ind:state state_ref="state_var_accounts_passwords_pam_faillock_deny_value_upper_bound" />
-    <ind:state state_ref="state_var_accounts_passwords_pam_faillock_deny_value_lower_bound" />
+  <ind:textfilecontent54_test check="all" check_existence="all_exist" version="1"
+        id="test_accounts_passwords_pam_faillock_deny_system_pam_faillock_auth"
+        comment="One and only one occurrence is expected in auth section of system-auth">
+    <ind:object
+        object_ref="object_accounts_passwords_pam_faillock_deny_system_pam_faillock_auth"/>
   </ind:textfilecontent54_test>
 
-  <!-- step 1 and 4 object -->
-  <ind:textfilecontent54_object id="object_accounts_passwords_pam_faillock_lines_value_password-auth" comment="Get number of lines pam_unix defaults to" version="1">
-    <ind:filepath>/etc/pam.d/password-auth</ind:filepath>
-    <ind:pattern operation="pattern match">[\n][\s]*auth[\s]+\[[^\]]*default=([0-9]+)[^\]]*\][\s]+pam_unix\.so</ind:pattern>
-    <ind:instance datatype="int">1</ind:instance>
+  <ind:textfilecontent54_object version="1"
+        id="object_accounts_passwords_pam_faillock_deny_system_pam_faillock_account"
+        comment="Check common definition of pam_faillock.so in account section of system-auth">
+    <ind:filepath operation="pattern match">^/etc/pam.d/system-auth$</ind:filepath>
+    <ind:pattern operation="pattern match"
+                 var_ref="var_accounts_passwords_pam_faillock_deny_pam_faillock_account_regex"/>
+    <ind:instance datatype="int" operation="equals">1</ind:instance>
```

Comment:
The `instance` is about which object to collect. 
You should collect all objects found, not just the first, so his should be `greater than or equal`  1.

Reply:
Hi @yuumasato , thanks for reviewing. The reason to get only the first `instance` is because these (auth and account) regexes detect the expected `pam_faillock.so` entries in relation to the `pam_unix.so` entry. Since the number of `pam_unix.so` entries is already verified previously, it is not expected to get more than 1 instance here. Otherwise the overall result of the first "criteria group" would already impact in the final assessment. Therefore, I limited the number of instances collected here since they are not necessary and `probably` a very small performance improvement should be obtained not collecting unnecessary data, even if they are not found. I didn't calculated the times with or without, but looks reasonable to me. : )

Reply:
I see, the gain obtained is probably very small.
I think the instance can stay as it, but I'd change the test to `check_existence="only_one_exists"` as that is the intention of the test.

Reply:
Adjusted.